tasks/kernel_matrix_testing/compiler.py

```diff
@@ -114,10 +113,44 @@ def start(self) -> None:
         self.exec("echo conda activate ddpy3 >> /home/compiler/.bashrc", user="compiler")
         self.exec(f"install -d -m 0777 -o {uid} -g {uid} /go", user="root")
 
-
-def get_compiler(ctx: Context, arch: ArchOrLocal):
-    return CompilerImage(ctx, full_arch(arch))
-
-
-def all_compilers(ctx: Context):
-    return [get_compiler(ctx, arch) for arch in arch_mapping.values()]
+        self.prepare_for_cross_compile()
+
+    def ensure_ready_for_cross_compile(self):
+        res = self.exec("test -f /tmp/cross-compile-ready", user="root", warn=True)
+        if res is not None and not res.ok:
```

Comment:
I think res is None should be an error condition here

Reply:
Fixed